CMakeLists.txt

```diff
@@ -278,9 +278,14 @@ if(AVIF_CODEC_DAV1D)
     set(AVIF_SRCS ${AVIF_SRCS} src/codec_dav1d.c)
 
     if(AVIF_LOCAL_DAV1D)
-        set(LIB_FILENAME "${CMAKE_CURRENT_SOURCE_DIR}/ext/dav1d/build/src/libdav1d.a")
+        set(LIB_FILENAME "${CMAKE_CURRENT_SOURCE_DIR}/ext/dav1d/build/src/libdav1d${AVIF_LIBRARY_SUFFIX}")
         if(NOT EXISTS "${LIB_FILENAME}")
-            message(FATAL_ERROR "libavif: ${LIB_FILENAME} is missing, bailing out")
+            # On windows, meson will produce a libdav1d.a instead of the expected libdav1d.dll/.lib.
```

Comment:
For reference, it's not only about meson - any toolchain on Windows that is not MSVC will do this by default (i.e. CMake w/ GCC/Clang, Autotools...). Just use the Cmake variables that are correctly set up for the platform, e.g.: https://gitlab.kitware.com/cmake/cmake/-/blob/ee7479fb5a457fae95e24d3b0d206d6138e5a448/Modules/Platform/Windows-GNU.cmake#L18-27

Reply:
Thanks for the information.

> For reference, it's not only about meson - any toolchain on Windows that is not MSVC will do this by default (i.e. CMake w/ GCC/Clang, Autotools...).

Do you have any link to some documentation about that, so that I can replace `https://github.com/mesonbuild/meson/issues/8153`?

>  Just use the CMake variables that are correctly set up for the platform, e.g.: https://gitlab.kitware.com/cmake/cmake/-/blob/ee7479fb5a457fae95e24d3b0d206d6138e5a448/Modules/Platform/Windows-GNU.cmake#L18-27

Not sure I follow, are you suggesting to use `CMAKE_FIND_LIBRARY_PREFIXES` and `CMAKE_FIND_LIBRARY_SUFFIXES`? Because looking for `libdav1d${AVIF_LIBRARY_SUFFIX}` did not work.

Reply:
> Not sure I follow, are you suggesting to use `CMAKE_FIND_LIBRARY_PREFIXES` and `CMAKE_FIND_LIBRARY_SUFFIXES`? Because looking for `libdav1d${AVIF_LIBRARY_SUFFIX}` did not work.

Hm, given [it's set up correctly already](https://github.com/AOMediaCodec/libavif/blob/cd03eef10a7cc6081ee98e3911e21d977a2bf17b/CMakeLists.txt#L73-L79), I'm not sure what your problem actually is now... Maybe the `lib` prefix should not be assumed and one should also use `${AVIF_LIBRARY_PREFIX}dav1d${AVIF_LIBRARY_SUFFIX}` like for the other libraries?

Reply:
> Maybe the `lib` prefix should not be assumed and one should also use `${AVIF_LIBRARY_PREFIX}dav1d${AVIF_LIBRARY_SUFFIX}` like for the other libraries?

From what I understood, `meson --default-library=static` will generate `libdav1d.a` instead of `dav1d.lib`.
A similar configuration issue with libgav1 is likely, because it does not use the `${AVIF_LIBRARY_PREFIX}` either.

> Is local dav1d not built using CMake?

No: https://github.com/AOMediaCodec/libavif/blob/0311709e9b7b8c46aace942e6b21207f48967e4c/ext/dav1d.cmd#L20

Reply:
Ah, ok, so it is meson + MSVC specific after all, sorry!

In that case, I guess you could iterate through the matrix of `CMAKE_FIND_LIBRARY_PREFIXES` and `CMAKE_FIND_LIBRARY_SUFFIXES`, or just handle this one known corner case like this.

Reply:
Sure. I guess `CMAKE_FIND_LIBRARY_PREFIXES` and `CMAKE_FIND_LIBRARY_SUFFIXES` can be combined with [`find_library()`](https://cmake.org/cmake/help/latest/command/find_library.html#command:find_library) in another PR.